libraries/chain/block_header_state.cpp

```diff
@@ -79,6 +77,30 @@ const finalizer_policy& block_header_state::get_last_proposed_finalizer_policy()
    return *active_finalizer_policy;
 }
 
+// The last pending finalizer policy if none pending then the active finalizer policy
+// Used to populate last_pending_finalizer_policy_digest which is expected to be the highest generation pending
+const finalizer_policy& block_header_state::get_last_pending_finalizer_policy() const {
+   if (!finalizer_policies.empty()) {
+      [[maybe_unused]] auto highest_pending_generation = [this]() {
+         finalizer_policy_ptr highest;
+         for (auto ritr = finalizer_policies.rbegin(); ritr != finalizer_policies.rend(); ++ritr) {
+            if (ritr->second.state == finalizer_policy_tracker::state_t::pending) {
+               if (!highest || highest->generation < ritr->second.policy->generation)
+                  highest = ritr->second.policy;
+            }
+         }
+         return highest;
+      };
+      for (auto ritr = finalizer_policies.rbegin(); ritr != finalizer_policies.rend(); ++ritr) {
+         if (ritr->second.state == finalizer_policy_tracker::state_t::pending) {
+            assert(highest_pending_generation() == ritr->second.policy);
+            return *ritr->second.policy;
+         }
+      }
+   }
+   return *active_finalizer_policy;
+}
```

Comment:
Strange that these (Kevin's changes) show up in your PR. Probably need to pull main to your branch.

Reply:
Yes, this PR depends on Kevin's PR, which I merged in here. I'll make sure to keep it up to date until it is merged.

Reply:
Oh, my bad, for some reason I thought Kevin's PR was already merged. That explains it!